Improve lang lib Module.md fie (#18855)
ballerina-platform/ballerina-lang: langlib/lang.map/src/main/ballerina/src/lang.map/Module.md

## Module Overview

This module provides lang library map operations defined by the language specification 2019R3. All the functions listed

> below are supported by the `map` type. Records do not support any functions which violate their respective inherent 

Reviewer: ```suggestion
in this module are supported by mapping types. Records do not support any functions that violate their inherent 
```

Author: Can’t say mapping types since records only support a subset of the functions..

Author: Can’t say mapping types since records only support a subset of the functions..

Reviewer: But saying `map` types might also be misleading, right? It's like they are only supported with `map`-typed variables. Then we talk about records, and there's a disconnect.

How about,
```suggestion
in this module are supported by mapping types, as long as the function does not violate the inherent type.
```

Functions like remove can also fail on maps, if immutable.

> types. e.g., removing a required field of a record

Reviewer: ```suggestion
types. e.g., removing a required field of a record.
```
